src/lib.rs

```diff
@@ -177,12 +177,12 @@ impl SeatRef {
     /// poll the libseat connection for events that need to be dispatched.
     ///
     /// Returns a pollable fd on success.
-    pub fn get_fd(&mut self) -> Result<RawFd, Errno> {
+    pub fn get_fd(&mut self) -> Result<BorrowedFd, Errno> {
         let fd = unsafe { sys::libseat_get_fd(self.0.as_mut()) };
         if fd == -1 {
             Err(errno())
         } else {
-            Ok(fd)
+            Ok(unsafe { BorrowedFd::borrow_raw(fd) })
```

Comment:
I think `get_fd` should be correct, but looking at how libseat works, returning an `OwnedFd` in `open_device` (like I did here) isn't right...

Returning a `BorrowedFd` may be okay, since `close_device` takes `&mut self`, so the `BorrowedFd` would be invalid. Though if `open_device` also takes `&mut self`, that may be awkward since the caller probably wants to keep a reference to the fd...

Or it could use some kind of wrapper type instead of `OwnedFd`/`BorrowedFd`. Perhaps if `close_device` took this wrapper type by value, it could provide a way to have an `AsFd` value without a lifetime bound that won't be closed while that value exists.